### ISSUE TITLE
Mac support

### ISSUE THREAD

**Issue by funatsufumiya** [mutouyun/cpp-ipc#130]
I'd like to know why macOS is not supported here. Are there any technical reasons?

**Comment by mutouyun** [comment]
No, the main reason is that I personally don't have a MacOS environment for development and testing.

**Comment by funatsufumiya** [comment]
I see. If any tests or checks are needed, maybe everyone helps you :)

**Comment by mutouyun** [comment]
I remember someone helped me with the macOS pr before, if you test on macOS and find compilation or running problems, you can also directly bring the pr to me~

**Comment by cristian64** [comment]
With the added support for FreeBSD, does it make any easier to add support for macOS?
